strongbox-resources/strongbox-common-resources/src/main/resources/logback-spring.xml

```diff
@@ -8,88 +8,152 @@
-
-                <encoder>
-                    <charset>UTF-8</charset>
-                    <pattern>%d{HH:mm:ss.SSS dd-MM-yyyy} | %-5.5p | %-20.20t | %-50.50logger{50} | %m%n</pattern>
-                </encoder>
-            </appender>
-        </sift>
-    </appender>
-
-    <!--
-        Please read the following article to make yourself familiar with logback configuration details:
-        https://logback.qos.ch/manual/configuration.html
-    -->
-
-    <!-- Strongbox packages -->
-    <logger name="org.carlspring.strongbox" level="INFO"/>
-    <logger name="org.carlspring.strongbox.security.authentication" level="INFO"/>
-    <logger name="org.carlspring.strongbox.security.vote" level="INFO"/>
-    <logger name="org.carlspring.strongbox.data" level="WARN"/>
-    <logger name="org.carlspring.strongbox.data.tx" level="WARN"/>
-    <logger name="org.carlspring.strongbox.data.criteria" level="INFO"/>
-    <logger name="org.carlspring.strongbox.data.server" level="INFO"/>
-    <!-- Strongbox packages -->
-
-    <!-- Third-party packages -->
-    <logger name="liquibase" level="INFO"/>
-
-    <logger name="ch.qos.logback" level="ERROR"/>
-
-    <logger name="com.orientechnologies.orient" level="ERROR"/>
-
-    <logger name="org.apache.commons.beanutils" level="ERROR"/>
-    <logger name="org.apache.http" level="ERROR"/>
-
-    <logger name="org.glassfish.jersey" level="ERROR"/>
-
-    <logger name="org.quartz" level="INFO"/>
-
-    <logger name="org.reflections" level="ERROR"/>
-
-    <logger name="org.springframework" level="ERROR"/>
-    <logger name="org.springframework.web" level="INFO"/>
-
-    <logger name ="org.eclipse.jetty" level="INFO" />
-    <logger name ="jndi"  level="INFO" />
-    
-    <logger name ="liquibase.parser.core.xml"  level="INFO" />
-    <logger name ="javax.xml.bind" level="INFO" />
-    <logger name ="javax.xml.bind" level="INFO" />
-    <logger name="org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping" level="INFO" />
-    <!-- Third-party packages -->
-
-    <root level="INFO">
-        <appender-ref ref="CONSOLE"/>
-        <appender-ref ref="FILE"/>
-    </root>
+    <springProperty scope="context" name="debug" source="logging.debug" defaultValue="false"/>
+    <springProperty scope="context" name="console" source="logging.console" defaultValue="true"/>
+    <springProperty scope="context" name="file" source="logging.file" defaultValue="true"/>
+
+    <if condition="${console:-true}">
+    <then>
+        <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
+            <filter class="org.carlspring.strongbox.log.CronTaskContextFilter"/>
+    
+            <encoder>
+                <pattern>%d{HH:mm:ss.SSS dd-MM-yyyy} | %-5.5p | %-20.20t | %-50.50logger{50} | %m%n</pattern>
+            </encoder>
+        </appender>
+        <root>
+            <appender-ref ref="CONSOLE"/>
+        </root>
+    </then>
+    </if>
+    
+    <if condition="${file:-true}">
```

Comment:
Wow! When did they add logical configurations to Logback?

Cool stuff!

![image](https://user-images.githubusercontent.com/1436265/59312531-65fbbd00-8ca5-11e9-9cf4-eb1a9dd494d7.png)

(Somebody's been thinking REAL HARD!) :laughing:

Reply:
yeah, we don't need `logback-spring-debug.xml` now